llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -427,6 +428,9 @@ int main(int argc, char **argv) {
     // Register required analysis
     MAM.registerPass([&] { return PassInstrumentationAnalysis(); });
     RunSpecConst.addPass(SCP);
+    if (!DoSplit)
+      // This pass deletes unreachable globals. Code splitter runs it later.
```

Comment:
It would be great if we add a test with spec constants handling and split both.

Reply:
This does not really apply to this fix. But this is important missing test case - agree, so I'll add a test.